### PR TITLE
Alias uniquifier

### PR DESCRIPTION
**Pull request by kimpepper**
This splits out the pathauto_alias_uniquify() method and pathauto_is_alias_reserved() into a new AliasUniquifier service.

### CODE REVIEW

File: pathauto.inc [hunk 1]
```diff
@@ -88,26 +88,12 @@ function _pathauto_existing_alias_data($source, $language = Language::LANGCODE_N
  *   A string with the path source.
  * @param string $langcode
  *   A string with a language code.
+ *
+ * @deprecated Use \Drupal::service('pathauto.alias_uniquifier')->uniquify()
```
Comment (Berdir) [inline pathauto.inc:92]: We changed so much that I don't really care about leaving deprecated functions in, I'm ok with just dropping this if we no longer need it. Want to kill pathauto.inc asap.

File: src/AliasStorageHelperInterface.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * @file
+ * Contains ${NAMESPACE}\\${NAME}
```
Comment (Berdir) [inline src/AliasStorageHelperInterface.php:4]: :)

I have the same problem with my phpstorm template, that used to work at some point..

**Comment by Berdir** [comment]
Looks nice, want to rebase the pull request instead of merging so the history is cleaner? Doesn't matter much, not sure if it will get merged in in the end.

**Comment by kimpepper** [comment]
> Looks nice, want to rebase the pull request instead of merging so the history is cleaner? 

OK. I'll close this issue and open up a new one.

**Pull request closed without merging**